File: scripts/rename_imports.py
import os
import glob

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as file:
        content = file.read()
    
    new_content = content.replace('@/lib/mock-data', '@/lib/api')
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as file:
            file.write(new_content)
        print(f"Updated {filepath}")

for root, _, files in os.walk('.'):
    for name in files:
        if name.endswith('.ts') or name.endswith('.tsx'):
            filepath = os.path.join(root, name)
            replace_in_file(filepath)

File: scripts/rename_imports.js
const fs = require('fs');
const path = require('path');

function replaceInFile(filepath) {
    const content = fs.readFileSync(filepath, 'utf8');
    const newContent = content.replace(/@\/lib\/mock-data/g, '@/lib/api');
    if (newContent !== content) {
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Updated ${filepath}`);
    }
}

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        if (fs.statSync(filepath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.next' && file !== '.git') {
                walk(filepath);
            }
        } else {
            if (filepath.endsWith('.ts') || filepath.endsWith('.tsx')) {
                replaceInFile(filepath);
            }
        }
    }
}

walk('.');
